src/ErrorHandler.php

```diff
@@ -199,16 +199,21 @@ private function renderThrowableAndTerminate(Throwable $t): void
         if (!empty($this->workingDirectory)) {
             chdir($this->workingDirectory);
         }
-        // disable error capturing to avoid recursive errors while handling exceptions
+        // Disable error capturing to avoid recursive errors while handling exceptions.
         $this->unregister();
-        // set preventive HTTP status code to 500 in case error handling somehow fails and headers are sent
+        // Set preventive HTTP status code to 500 in case error handling somehow fails and headers are sent.
         http_response_code(Status::INTERNAL_SERVER_ERROR);
 
         echo $this->handle($t);
         $this->eventDispatcher?->dispatch(new ApplicationError($t));
 
         register_shutdown_function(static function (): void {
-            exit(1);
+            // Ensure exit(1) is called last after all other shutdown functions, even those added to the end.
+            register_shutdown_function(static function (): void {
```

Comment:
Looks like an infinite process to reach the last queue. Could it be simpler?

Reply:
No:

1. There's no way to tell current list of shutdown functions.
2. There's no way to unregister a shutdown function.
3. We can't use a custom shutdown queue, since we don't control third party code that may register shutdown functions.

Reply:
Could it be then be controllable?
new ErrorHandler($errorHandlerDepth = 2)

And 2 is the number of the shutdown function depth

Reply:
Done.